Compile warning on winbuilder

```
integer64.c: In function 'runif_integer64':
integer64.c:1015:15: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
 1015 |     while( (*((long long *) &ii)) == NA_INTEGER64){
      |              ~^~~~~~~~~~~~~~~~~~
integer64.c:1023:36: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
 1023 |     ret[i] = min + ( (long long)(*((unsigned long long *)(&ii))) % d);
      |                                   ~^~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

https://win-builder.r-project.org/Y2yv92Drbtjt/00install.out

--- a/src/integer64.c
+++ b/src/integer64.c
@@ -1010,18 +1010,22 @@ SEXP runif_integer64(SEXP n_, SEXP min_, SEXP max_){
   for (i=0; i<n; i++){
     ii.low = (unsigned int) floor(unif_rand()*4294967296);
     ii.high = (unsigned int) floor(unif_rand()*4294967296);
-//#pragma GCC diagnostic push
-//#pragma GCC diagnostic ignored "-Wstrict-aliasing"
+#ifdef __GNUC__
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wstrict-aliasing"
     while( (*((long long *) &ii)) == NA_INTEGER64){
-//#pragma GCC diagnostic pop
+#pragma GCC diagnostic pop
+#endif
       // xx optimisation opportunity: if we know endianess, we only need to replace one of the two
       ii.low = (unsigned int) floor(unif_rand()*4294967296);
       ii.high = (unsigned int) floor(unif_rand()*4294967296);
     }
-//#pragma GCC diagnostic push
-//#pragma GCC diagnostic ignored "-Wstrict-aliasing"
+#ifdef __GNUC__
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wstrict-aliasing"
     ret[i] = min + ( (long long)(*((unsigned long long *)(&ii))) % d);
-//#pragma GCC diagnostic pop
+#pragma GCC diagnostic pop
+#endif
   }
   PutRNGstate();
   UNPROTECT(1);

